scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -1131,6 +1132,64 @@ def _check_undeclared_props(self):
                      f"{self.edt.dts_path}, but is not declared in "
                      f"'properties:' in {self.binding_path}")
 
+    def _init_ranges(self):
+        # Initializes self.ranges
+        node = self._node
+
+        self.ranges = []
+
+        if "ranges" not in node.props:
+            return
+
+        child_address_cells = node.props.get("#address-cells")
+        parent_address_cells = _address_cells(node)
+        if not child_address_cells:
+            child_address_cells = 2 # Default value per DT spec.
+        else:
+            child_address_cells = child_address_cells.to_num()
+        child_size_cells = node.props.get("#size-cells")
+        if not child_size_cells:
+            child_size_cells = 1 # Default value per DT spec.
+        else:
+            child_size_cells = child_size_cells.to_num()
+
+        # Number of cells for one translation 3-tuple in 'ranges'
+        entry_cells = child_address_cells + parent_address_cells + child_size_cells
+
+        for raw_range in _slice(node, "ranges", 4*entry_cells,
+                "4*(<#address-cells> (= {}) + "
+                "<#address-cells for parent> (= {}) + "
+                "<#size-cells> (= {}))"
+                .format(child_address_cells, parent_address_cells,
+                    child_size_cells)):
+
+            range_ = Range()
+            range_.node = self
+            if child_address_cells == 0:
+                range_.child_bus_flags = None
+                range_.child_bus_addr = None
+            elif child_address_cells == 3:
+                # When used as PCIe ranges, the first entry can be some flags
```

Comment:
This seems to be special case code for PCIe, but there is no indication that the node is related to PCIe, so why is this here? I don't understand how it's safe to assume this in general.; nothing in the DT spec says that 3 is a magic value which means the first entry is always PCIe-specific flags, and this code runs for every node in the DT.

Reply:
(in ref to https://github.com/zephyrproject-rtos/zephyr/pull/38106#discussion_r718295619)
Yep I understand, but I don't see any other way to handle `#address-cells = 3`

Reply:
>  Yep I understand, but I don't see any other way to handle `#address-cells = 3`

Is there some other way the node marks itself as a PCIe controller? Some sort of property or node name convention?

Reply:
... and if not, perhaps we can get some extra hints from the bindings to use in edtlib. E.g. we can treat the node differently if and only if we see `bus: pci` (https://docs.zephyrproject.org/latest/guides/dts/bindings.html#bus) in its matching binding.

Reply:
@mbolivar-nordic OK I did that.

I only strictly parsed Ranges in edtlib (so address-cells generates 92bit numbers in python, but it's ok) and fixed the tests accordingly. I also passed the cells count for each part of a range, because it's used in the case of PCIe to know the width of the first range entry to extract the flags.

Then I simply used the bus bindings property in gen_defines to extract the flags.